doc/memos/rdm0.md

```diff
@@ -0,0 +1,148 @@
+about architectural design decisions, code structure etc. Other memos might
+also describe other aspects of RIOT activity, including, but not limited to,
+RIOT community processes, position with respect to some related external
+technical debates etc.
+
+
+This memo specifies a format for such memos and the process of discussing their 
+content before publication. Format and processes are somewhat inspired by 
+[IETF](http://ietf.org/) processes and documents such as Request For Comments 
+(RFC).
+
+# 2. RIOT Developers Memo Format
+
+RIOT Developer Memos (RDM) are written in English, using the Markdown (.md) 
+format, with less than 80 characters per line. Each memo has a unique 
+identification number indicated in its header, 
+e.g. this memo is RDM0.
+
+The structure of a memo should contain:
+
+1. a preamble as defined below,
+2. the content of the memo structured as needed,
+3. a postamble as defined below.
+
+The preamble of the memo should be structured as such:
+
+1. a header formatted as the header of this memo,
+2. a short abstract,
+3. a status for the memo stating its license and the category of the memo (e.g. 
+that it is a consensus document),
+4. optionally, a terminology section which recalls/disambiguates definition of 
+terms that are key to the memo.
+
+The core content of the memo is structured as deemed necessary. However, during 
+the review process, care must be taken to follow the below principles as much
+as possible:
+
+1. one memo = one problem/aspect,
+2. start by crisply explaining and motivating the one problem/aspect the memo 
+targets,
+3. memos are expected to be of normative and durable nature, hence: keep a memo 
+as concise as possible without impairing clarity, and factor out content that 
+would be likely to be updated too often.
+
+The postamble of the memo should be structured as such:
+
+1. (optional) acknowledgements section thanking contributors other than the 
+authors,
+2. a section with the main references that relate to the content, with a strong
+preference for permanent, self-contained references,
+3. the list of revisions so far and corresponding changelog,
+4. contact information to reach the author(s).
+
+
+# 3. Memo Publishing and Maintenance Process
+
+Each memo is proposed as a pull-request (PR) and discussed the same way RIOT
+code PRs are processed on GitHub.
+
+
+Once proposed, a memo must have received an ACK by at least 3 maintainers other 
+than the author(s) of the memo, before its publication. 
+
+
+At this stage, the RDM is assigned the next available RDM number, and merged in 
+the master branch of RIOT. Once merged into the master branch of RIOT, a memo 
+is considered published.
+
+Small changes/clarifications may be subsequently PRed to improve a memo, using 
+the same RDM number. In this case, a (monotically increasing) revision number 
+is issued and a changelog is added -- see Revision section in postamble below.
```

Comment:
I think some mechanism of alert should be encouraged so that people know when they _need_ to update a memo so that it's kept accurate. For example, design documents, when the design is changed in the code.
What I mean is something like maybe a link to the RDM in the code where it applies. Or, in the .md document where it applies.

Reply:
@danpetry see my comment above.
an RDM is not necessarily tied to a concurrent, specific code contribution PR.
To the contrary, RDMs should target more generic documentation of positioning, known pitfalls and best practices.
Specifying exactly *when* to update or deprecate an RDM is thus tricky.
I think we should leave this unspecified, and let the community decide in an ad hoc fashion, case by case.

Reply:
Just to clarify: here I'm talking about RDMs that _are_ tied to a code contribution PR. I understand that they can be contributed on a range of subject areas, including process, best practices, technical and non-technical etc.
I think I see what you mean, though. Do you mean that even when they're technical, they would be rarely (if at all) tied to a specific module of code. E.g. design of abstraction layers or whole subsystems like OTA for example?
I can see that people might want to contribute an RDM for a module design, and then in that case it would be coupled and would need to be changed if the code changes.

Reply:
OK. See my above comment: we should probably aim to merge RDMs that document "stable" design/architecture stuff only when that stuff is either already merged, or just about to be merged.
Basically: I think we should avoid over-specifying when to write/modify an RDM (ref. TinyOS).
But I agree with you that we should not prevent people from PRing RDMs early, if it helps, in some cases.
Makes sense?

Reply:
Maybe we should discuss this more when you're in - disagree that they should aim only to document stable architectures. I think it would be a great use for them to allow people to work on architectures together, across the world. What happens if someone on the other side of the world contributes a key feature which we know is broken by design? Or if a group of people all have expertise in a topic and want to design a feature together.
Agree that we shouldn't _specify_ this. But I think we should point it out as a good use for them, and probably recommend it too.